### DISCUSSION TITLE
Internacionalização em vue

### DISCUSSION THREAD

**Discussion by mateusortiz** [frontendbr/forum#1509] [Perguntas e Respostas]
Opa Galera,

_Me apresentando rapidamente sou frontend dev de inovação de uma startup em Curitiba_, e estamos pensando em internacionalização(i18n), o produto é um erp na núvem, e precisamos deixar fácil esse controle da internacionalização no nosso painel administrativo do sistema, para qualquer pessoa conseguir alterar esse localização de idioma.

Sei que já existe uma pergunta aqui sobre internacionalização, mas queria ver os pontos usando o vue, e se faz mas sentido no backend ou no frontend

**Comment by brunosantanarj** [comment]
[Aqui](https://nuxtjs.org/examples/i18n) tem um exemplo de internacionalização e Server Side Rendering com Vue. 

Há uns 2 anos atrás usei essa abordagem de SSR com Nuxt e um dos requisitos era a internacionalização, funcionou bem na época.

Se faz mais sentido fazer isso no backend ou frontend pode depender de qual padrão está usando para renderizar suas páginas.

**Comment by nandomoreirame** [comment]
acredito que qualquer formatação visual para o cliente (seja linguagem, datas, moedas, etc) fica sob responsabilidade do front-end... é que depende bastante de como tá estruturada a aplicação tbm, mas não acho certo fazer requisições na API para traduzir textos... pensando em aplicações que usam MVC (RoR, Laravel, e outros) o i18n fica sob responsabilidade da View, mesmo sendo renderizada no server-side 😄 

#meus10centavos

**Comment by nelsonmfinda** [comment]
> acredito que qualquer formatação visual para o cliente (seja linguagem, datas, moedas, etc) fica sob responsabilidade do front-end... é que depende bastante de como tá estruturada a aplicação tbm, mas não acho certo fazer requisições na API para traduzir textos... pensando em aplicações que usam MVC (RoR, Laravel, e outros) o i18n fica sob responsabilidade da View, mesmo sendo renderizada no server-side 
> 
> #meus10centavos

@nandomoreirame Pelo que entendi, seria uma boa opção implementar ***internacionalização*** via ***i18n*** no _frontend_ certo?

**Comment by ftonato** [comment]
Hey there,

Atualmente trabalho em um ambiente onde as aplicações são em sua maioria desenvolvidas com Vue, pensando no _custo X benefício X tempo_, acabamos por usar o [vue-i18n](https://github.com/kazupon/vue-i18n) que é extremamente fácil de ser implementado mesmo em aplicações que já existiam e não tinham intenacionalização (que é o seu caso).

**Comment by alvarengathomas** [comment]
Temos diversos projetos em Vue e todos eles tomam conta da internacionalização sem depender do backend, utilizamos [vue-i18n](https://github.com/kazupon/vue-i18n)

**Comment by kavalcante** [comment]
Galera, em aplicações de grande porte. Vocês ainda lidam com a internacionalização via arquivo JSON? O vue-i18n já lida com implementações buscando em BD?

**Comment by lucasferreiralimax** [comment]
> Galera, em aplicações de grande porte. Vocês ainda lidam com a internacionalização via arquivo JSON? O vue-i18n já lida com implementações buscando em BD?

Creio que o vue-i18n sempre foi agnóstico com a forma que implementa os arquivos, seja eles em formato json, object js ou em uma requisição no BD.
